### PR TITLE
editoast: migrate POST /rolling_stock and implement model for rolling_stock

### LINKED ISSUE #3499: migrate POST /rolling-stock/ endpoint to editoast
# Description
Migrate `POST /rolling-stock/` endpoint to editoast. This endpoint is used to create a new rolling-stock

# Goal
Drop API service.

# Acceptance Criteria
- [ ] the frontend use the editoast endpoint and the user does not see any difference
- [ ] the endpoint doesn't exist anymore in API

### PR DESCRIPTION
**Pull request by clarani**
closes #3499

### CODE REVIEW

File: editoast/src/schema/rolling_stock.rs
Comment (younesschrifi) [inline editoast/src/schema/rolling_stock.rs:1]: For me, you should move the code from this file  to`/models/rolling_stocks.rs`

File: editoast/openapi.yaml [hunk 1]
```diff
@@ -1029,6 +1029,19 @@ paths:
               schema:
                 $ref: "#/components/schemas/LightRollingStock"
 
+  /rolling_stock/:
+    post:
+      tags:
+        - rolling_stock
+      summary: Create a rolling stock
```
Comment (younesschrifi) [inline editoast/openapi.yaml:1038]: You should add the content

**Review by younesschrifi — CHANGES_REQUESTED** [review]
Nice PR :)

File: editoast/src/schema/rolling_stock.rs
Comment (clarani) [reply editoast/src/schema/rolling_stock.rs:1]: I will move the code of LightRollingStock in a following PR, here is the ticket : https://github.com/DGEXSolutions/osrd/issues/3642

**Review by younesschrifi — DISMISSED** [review]

File: editoast/src/models/rolling_stock.rs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                .await
+                .unwrap()
+                .is_some()
+        );
+
+        // delete a rolling stock
+        assert!(RollingStockModel::delete(db_pool.clone(), rolling_stock_id)
+            .await
+            .unwrap());
+    }
```
Comment (Tguisnet) [inline editoast/src/models/rolling_stock.rs]: You could use transactions (same for all the tests), and maybe split these tests.

File: editoast/src/views/rolling_stocks.rs [hunk 3]
```diff
@@ -40,16 +50,77 @@ async fn get_livery(db_pool: Data<DbPool>, path: Path<(i64, i64)>) -> Result<Htt
...
+    pub rolling_resistance: DieselJson<RollingResistance>,
+    pub loading_gauge: String,
+    pub metadata: DieselJson<RollingStockMetadata>,
+    pub power_restrictions: Option<JsonValue>,
+}
+
+impl From<RollingStockCreateForm> for RollingStockModel {
+    fn from(rolling_stock: RollingStockCreateForm) -> Self {
+        RollingStockModel {
+            id: None,
```
Comment (Tguisnet) [inline editoast/src/views/rolling_stocks.rs]: You could implement the Trait Default, and use it.

**Review by Tguisnet — CHANGES_REQUESTED** [review]
Good PR.

- We should use transactions for tests involving the DB.
- Maybe we should create sub module for Rolling Stock ?
- Should we rename osrd_infra_rollingstock into osrd_rollingstock (same for projects, studies, ...) ?

File: editoast/src/views/rolling_stocks.rs (hunk 3 above)
Comment (clarani) [reply editoast/src/views/rolling_stocks.rs]: Done

File: editoast/src/models/rolling_stock.rs (hunk 2 above)
Comment (clarani) [reply editoast/src/models/rolling_stock.rs]: Done

**Review by Tguisnet — APPROVED** [review]

**Pull request merged**